packages/db/src/query/device.ts

```diff
@@ -168,3 +168,67 @@ async function copyToHistoryTable(
         powerEstimation: device.powerEstimation,
     });
 }
+
+export async function saveDeviceToPeakDb(sensorDataSequenceId: string, deviceName: string, userId: string) {
+    try {
+        const deviceResult = await db
+            .select()
+            .from(device)
+            .where(and(eq(device.name, deviceName), eq(device.userId, userId)))
+            .limit(1);
+
+        if (deviceResult.length === 0) {
+            throw new Error("Gerät nicht gefunden.");
+        }
+
+        await db.insert(deviceToPeak).values({
+            deviceId: deviceResult[0].id,
+            sensorDataSequenceId,
+        });
+    } catch (error) {
+        console.error("Error saving device to peak:", error);
+        throw new Error("Failed to save device to peak.");
+    }
+}
+
+export async function createStandardDevicesIfNotExist(userId: string) {
+    const existingDevices = await getDevicesByUser(userId);
+    const existingDeviceNames = new Set(existingDevices.map((device) => device.name));
+
+    const deletedDevices = await db.select().from(deviceHistory).where(eq(deviceHistory.userId, userId));
+    const deletedDeviceNames = new Set(deletedDevices.map((device) => device.name));
+
+    const standardDevices = [
+        { name: "Waschmaschine", category: "washingMachine" },
+        { name: "Spülmaschine", category: "dishwasher" },
+        { name: "Gefrierschrank", category: "freezer" },
+        { name: "Kühlschrank", category: "fridge" },
+        { name: "Mikrowelle", category: "microwave" },
+    ];
+
+    for (const device of standardDevices) {
+        if (!existingDeviceNames.has(device.name) && !deletedDeviceNames.has(device.name)) {
+            await createDevice({
+                name: device.name,
+                category: device.category,
+                userId: userId,
+            });
+        }
+    }
+}
+
+export async function getPeaksWithoutDevices(peaks: { id: string }[]) {
+    const peaksToClassify: { id: string }[] = [];
+    for (const peak of peaks) {
+        const existingAssignments = await db
```

Comment:
Die Performance ist hier nicht so toll (gerade mit vielen Peaks). Besser wäre aus meiner Sicht, einen Inner Join durchzuführen.

In SQL also:
``` SQL
SELECT DISTINCT sensor_data_sequence.id  FROM sensor_data_sequence
INNER JOIN deviceToPeak ON deviceToPeak.sensor_data_sequence_id = sensor_data_sequence.id
WHERE sensor_data_sequence.id IN (1, 2, 3, 4)
```
Hinweis: 1,2,3,4 sind hier die peak IDs, die man reingegeben hat.

Dann hat man alle zugewiesenen Peaks. Dann kann man in SQL noch ein Mengen-Differenz bilden, um die eigentlich gewollten Peaks zu kriegen (hier der Einfachheit halber weggelassen)

Ich weiß leider nicht, wie man das ganze mit Drizzle macht, müsste man dann schauen